### PR TITLE
SERVER-84079 Introduce query settings Genny performance tests

### PR DESCRIPTION
**Pull request by denis631**
**Jira Ticket:** SERVER-84079

**Whats Changed:**  
Introduce a findOne by index queries with and without query settings to measure the performance

### CODE REVIEW

**Comment by denis631** [comment]
[🌲 sys-perf with genny build (query settings variants only)](https://spruce.mongodb.com/version/65cf7f5e306615a113cccde6/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

File: src/workloads/query/QuerySettings.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: findOne
+          OperationCommand:
+            Filter: *HitFilter
```
Comment (mmalinauskas) [inline src/workloads/query/QuerySettings.yml:86]: Even if the pseudo-random number generator seed is fixed for Genny test run, it is not clear if the filter matches a document. That likely can become very important if we want to compare results between no query settings, matching and non-matching query settings due to low resource consumption of the "find" command. I find this randomness and uncertainty unnecessary in this already messy world :).
I would suggest explicitly testing a scenario in which at least one document matches.

File: src/workloads/query/QuerySettings.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection querySettingsColl
+  DocumentCount: &DocumentCount 100000
```
Comment (mmalinauskas) [inline src/workloads/query/QuerySettings.yml:18]: Is there a value of having so many documents in the test collection? Insert of 1000s of docs takes time.

File: src/workloads/query/QuerySettings.yml [hunk 3]
```diff
@@ -0,0 +1,164 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  Measure the performance for find command with-/out query settings as well as checking both hit and miss cases.
```
Comment (mmalinauskas) [inline src/workloads/query/QuerySettings.yml:4]: "find" command

Comment (mmalinauskas) [inline src/workloads/query/QuerySettings.yml:4]: "both hit and miss cases" -> both query settings hit and miss cases"

Comment (mmalinauskas) [inline src/workloads/query/QuerySettings.yml:4]: with-/out -> "with or w/o"

**Review by mmalinauskas — COMMENTED** [review]
Looks good @denis631!

File: src/workloads/query/QuerySettings.yml (hunk 2 above)
Comment (denis631) [reply src/workloads/query/QuerySettings.yml:18]: Would you propose inserting 1000 documents instead?

**Review by denis631 — COMMENTED** [review]
Thank you for your review, Mindaugas!

File: src/workloads/query/QuerySettings.yml (hunk 1 above)
Comment (mmalinauskas) [reply src/workloads/query/QuerySettings.yml:86]: That's correct.

File: src/workloads/query/QuerySettings.yml (hunk 2 above)
Comment (mmalinauskas) [reply src/workloads/query/QuerySettings.yml:18]: The absolute minimum, whatever it is.

**Comment by denis631** [comment]
[🌲 genny run](https://spruce.mongodb.com/version/65d8c5463066157819014382/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

**Pull request closed without merging**